plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -825,29 +829,67 @@ protected boolean isSupportedJoinCondition(JdbcJoinCondition joinCondition)
         return true;
     }
 
-    private static ColumnMapping charColumnMapping(int charLength)
+    private static ColumnMapping charColumnMapping(int charLength, boolean withCollate)
     {
         if (charLength > CharType.MAX_LENGTH) {
-            return varcharColumnMapping(charLength);
+            return varcharColumnMapping(charLength, withCollate);
         }
         CharType charType = createCharType(charLength);
         return ColumnMapping.sliceMapping(
                 charType,
                 charReadFunction(charType),
-                charWriteFunction(),
-                POSTGRESQL_CHARACTER_PUSHDOWN);
+                withCollate ? charWriteFunctionWithCollate() : charWriteFunction(),
+                withCollate ? FULL_PUSHDOWN : POSTGRESQL_CHARACTER_PUSHDOWN);
+    }
+
+    private static SliceWriteFunction charWriteFunctionWithCollate()
```

Comment:
This is exact same as `varcharWriteFunctionWithCollate` (because we don't pad char values to their actual length)

you can call it `stringWriteFunctionWithCollate` and have common implementation

Reply:
Fixed.